flamingpy/examples/decoding.py

```diff
@@ -24,104 +24,106 @@
+        decoder = {"outer": "MWPM"}
+
+    # Drawing options
+    node_colors = "state" if noise == "cv" else False
+    dw = {
+        "show_nodes": True,
+        "color_nodes": node_colors,
+        "label": None,
+        "legend": True,
+        "title": True,
+        "display_axes": True,
+        "label_edges": True,
+        "label_cubes": False,
+        "label_boundary": False,
     }
-    decoder = {"inner": "basic", "outer": "MWPM"}
-
-if noise == "dv":
-    # i.i.d Pauli Z errors with probability p_Z
-    p_Z = 0.02
-    IidNoise(RHG_code, p_Z).apply_noise()
-    weight_options = {"method": "unit"}
-    decoder = {"outer": "MWPM"}
-
-# Drawing options
-node_colors = "state" if noise == "cv" else False
-dw = {
-    "show_nodes": True,
-    "color_nodes": node_colors,
-    "label": None,
-    "legend": True,
-    "title": True,
-    "display_axes": True,
-    "label_edges": True,
-    "label_cubes": False,
-    "label_boundary": False,
-}
-
-# Manual decoding (to plot intermediate results).
-dec.assign_weights(RHG_code, **weight_options)
-for ec in RHG_code.ec:
-    G_match = dec.build_match_graph(RHG_code, ec)
-    matching = G_match.min_weight_perfect_matching()
-    G_stabilizer = getattr(RHG_code, ec + "_stab_graph")
-
-    # An integer label for each nodes in the stabilizer and matching graphs.
-    # This is useful to identify the nodes in the plots.
-    node_labels = {node: index for index, node in enumerate(G_stabilizer.nodes())}
-
-    # The draw_dec_graph function requires the networkx backend. Most backends implement
-    # the to_nx() method to perform the conversion if needed.
-    RHG_code.draw_stabilizer_graph(
-        ec, title=ec.capitalize() + " stabilizer graph", node_labels=node_labels
+
+    # Manual decoding (to plot intermediate results).
+    dec.assign_weights(RHG_code, **weight_options)
+    for ec_ in RHG_code.ec:
+        G_match = dec.build_match_graph(RHG_code, ec_)
+        matching = G_match.min_weight_perfect_matching()
+        G_stabilizer = getattr(RHG_code, ec_ + "_stab_graph")
+
+        # An integer label for each nodes in the stabilizer and matching graphs.
+        # This is useful to identify the nodes in the plots.
+        node_labels = {node: index for index, node in enumerate(G_stabilizer.graph)}
+
+        # The draw_dec_graph function requires the networkx backend. Most backends implement
+        # the to_nx() method to perform the conversion if needed.
+        G_stabilizer.draw(title=ec_.capitalize() + " stabilizer graph", node_labels=node_labels)
+        ax = viz.syndrome_plot(RHG_code, ec_, drawing_opts=dw, index_dict=node_labels)
+        viz.draw_matching_on_syndrome_plot(ax, matching, G_match)
+        if len(G_match.graph):
+            G_match.draw(title=ec_.capitalize() + " matching graph", node_labels=node_labels)
+        else:
+            print("\nMatching graph empty!\n")
+
+        if show:
+            plt.show()
+        else:
+            plt.close()
```

Comment:
I suggest we add a new argument, `draw=True`, and enclose this in "if draw:" .

Reply:
Done! Is this exactly what you meant?

Reply:
Yes, thanks!